components/automate-gateway/gateway/services.go

```diff
@@ -874,12 +874,15 @@ func (s *Server) authRequest(r *http.Request, resource, action string) (context.
 
 	projects := auth_context.ProjectsFromMetadata(md)
 
-	authzResp, err := s.authorizer.IsAuthorized(ctx, subjects, resource, action, projects)
+	newCtx, authorized, err := s.authorizer.IsAuthorized(ctx, subjects, resource, action, projects)
 	if err != nil {
+		// any error is handled as "permission denied" at the call sites, with the error text returned
 		return nil, errors.Wrap(err, "authz-service error")
 	}
-
-	// err is nil if authorization succeeded
-	// Note: if we need all the auth info, use auth_context.NewOutgoingContext
-	return auth_context.NewOutgoingProjectsContext(authzResp.Ctx()), authzResp.Err()
+	if authorized {
+		// Note: if we need all the auth info, use auth_context.NewOutgoingContext
+		return auth_context.NewOutgoingProjectsContext(newCtx), nil
+	}
+	return nil, errors.Errorf("subjects %v are not permitted to do action %v to resource %v in projects %v",
```

Comment:
Compare the text to components/automate-gateway/gateway/middleware/authv2/authv2.go, L71 and L76. Should we use the same verbiage? Not sure. Different verbiage allows an unambiguous grep in the log if needed. However, might at least one to be consistent on `%v` vs `%q`. Does `%v` deliver less escaped quotes...?